spec/models/article_spec.rb

```diff
@@ -1436,27 +1436,18 @@ def foo():
       end
 
       it "detects language for new articles" do
-        allow_any_instance_of(CLD3::NNetLanguageIdentifier).to receive(:find_language).and_return(double(
-          "LanguageOutcome", probability: 0.8, reliable?: true, language: "es"
-        ))
+        allow_any_instance_of(Languages::Detection)
+          .to receive(:call)
+          .and_return(:es)
```

Comment:
You really don't need to mock `any_instance` (and _really **should not**_ if you can at all avoid it). Because `article#detect_language` invokes `Languages::Detect.call`, you can mock the class-level method:

```ruby
  allow(Languages::Detection).to receive(:call).and_return(:es)
```

(As a bonus, you can remove the `#rubocop` bits.)

Reply:
Also, the way the language mock is layered in here is kind of hard to follow. I'm pretty sure you wind up testing the same behavior multiple times (1433 and 1438 are basically the same test) and not testing the actual interface.

What about something like this approach:

```ruby
describe "#detect_language" do
  let(:detected_language) { :bg }

  before do
    allow(Languages::Detection).to \ 
      receive(:call).and_return(detected_language)
  end
  
  it "should detect language using title and body" do
    [...]
    expect(Languages::Detection).to \
      receive(:call).with("#{title}. #{body_text}")
  end
  
  it "should not call detection when title and body_markdown are unchanged" do
    [...]
    expect(Languages::Detection).not_to \
      have_received(:call)
  end
end
```